test/notifications/display_test.dart

```diff
@@ -265,54 +289,224 @@ void main() {
+
+      final expectedTagComponent = 'dm:${message1.allRecipientIds.join(",")}';
+
+      await receiveFcmMessage(async, data1);
+      checkNotification(data1,
+        messageStyleMessages: [data1],
+        expectedIsGroupConversation: true,
+        expectedTitle: "${eg.otherUser.fullName} to you and 1 other",
+        expectedTagComponent: expectedTagComponent);
+
+      await receiveFcmMessage(async, data2);
+      checkNotification(data2,
+        messageStyleMessages: [data1, data2],
+        expectedIsGroupConversation: true,
+        expectedTitle: "${eg.thirdUser.fullName} to you and 1 other",
+        expectedTagComponent: expectedTagComponent);
+    })));
+
+    test('1:1 DM', () => runWithHttpClient(() => awaitFakeAsync((async) async {
       await init();
       final message = eg.dmMessage(from: eg.otherUser, to: [eg.selfUser]);
       await checkNotifications(async, messageFcmMessage(message),
         expectedIsGroupConversation: false,
         expectedTitle: eg.otherUser.fullName,
         expectedTagComponent: 'dm:${message.allRecipientIds.join(",")}');
-    }));
+    })));
+
+    test('1:1 DM: title updates when sender name changes', () => runWithHttpClient(() => awaitFakeAsync((async) async {
+      await init();
+      final otherUser = eg.user(fullName: 'Before');
+      final message1 = eg.dmMessage(from: otherUser, to: [eg.selfUser]);
+      final data1 = messageFcmMessage(message1);
+
+      final expectedTagComponent = 'dm:${message1.allRecipientIds.join(",")}';
+
+      await receiveFcmMessage(async, data1);
+      checkNotification(data1,
+        messageStyleMessages: [data1],
+        expectedIsGroupConversation: false,
+        expectedTitle: 'Before',
+        expectedTagComponent: expectedTagComponent);
 
-    test('self-DM', () => awaitFakeAsync((async) async {
+      otherUser.fullName = 'After';
+      final message2 = eg.dmMessage(from: otherUser, to: [eg.selfUser]);
+      final data2 = messageFcmMessage(message2);
+
+      await receiveFcmMessage(async, data2);
+      checkNotification(data2,
+        messageStyleMessages: [data1, data2],
+        expectedIsGroupConversation: false,
+        expectedTitle: 'After',
+        expectedTagComponent: expectedTagComponent);
+    })));
+
+    test('1:1 DM: conversation stays same when sender email changes', () => runWithHttpClient(() => awaitFakeAsync((async) async {
+      await init();
+      final otherUser = eg.user(email: 'before@example.com');
+      final message1 = eg.dmMessage(from: otherUser, to: [eg.selfUser]);
+      final data1 = messageFcmMessage(message1);
+
+      final expectedTagComponent = 'dm:${message1.allRecipientIds.join(",")}';
+
+      await receiveFcmMessage(async, data1);
+      checkNotification(data1,
+        messageStyleMessages: [data1],
+        expectedIsGroupConversation: false,
+        expectedTitle: otherUser.fullName,
+        expectedTagComponent: expectedTagComponent);
+
+      otherUser.email = 'after@example.com';
```

Comment:
Ah good catch — I missed this in my review of the previous revision :slightly_smiling_face: